File: android/app/src/main/java/com/adqmobile/socialnetwork/domain/post/usecases/GetAllPostsUseCase.kt
package com.adqmobile.socialnetwork.domain.post.usecases;

import com.adqmobile.socialnetwork.domain.post.Post
import com.adqmobile.socialnetwork.domain.post.PostRepository

class GetAllPostsUseCase(private val postRepository: PostRepository) {

    suspend fun run(): Array<Post> {
        return postRepository.retrieveAllPosts()
    }
}

File: flutter/lib/domain/post/usecases/getAllPostsUseCase.dart
import '../postRepostiory.dart';

import '../post.dart';

class GetAllPostsUseCase {
  PostRepository postRepository;

  GetAllPostsUseCase(PostRepository postRepository) {
    this.postRepository = postRepository;
  }

  Future<List<Post>> run() {
    return postRepository.retrieveAllPosts();
  }
}